common/controllers/framework/confirm-assessment.js

```diff
@@ -17,6 +19,28 @@ class ConfirmAssessmentController extends FormWizardController {
     res.redirect(`/move/${moveId}`)
   }
 
+  async saveValues(req, res, next) {
+    const { assessment } = req
+    const services = {
+      'person-escort-record': req.services.personEscortRecord,
+      'youth-risk-assessment': req.services.youthRiskAssessment,
+    }
+
+    try {
+      await services[assessment.framework.name].confirm(assessment.id)
+
+      next()
+    } catch (err) {
+      const apiErrorCode = get(err, 'errors[0].code')
+
+      if (err.statusCode === 422 && apiErrorCode === 'taken') {
+        return this.successHandler(req, res)
```

Comment:
If I remember correctly, if we get a 422 and continue onto the next step, we'll show a 'taken' message on that screen?

So this code is only catching the known taken error, and other errors should show up as errors?

Reply:
Those are currently only picked up in the [save controllers](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/a45e499e51f00792e709644487d3468299200b20/app/move/app/new/controllers/save.js#L128-L153) are specific to creating a duplicate resource.

Although it might be better to follow that pattern of catching it in the `errorHandler` instead of directly when saving?

Also, its worth noting that this code isn't going to be `taken`, I'm waiting for @cwrw to confirm what the code will be. This error occurs when an assessment is already confirmed and we're attempting to patch it with the same status.